core/node/consensus/src/storage/store.rs

```diff
@@ -38,6 +38,7 @@ fn to_fetched_block(
         l1_gas_price: payload.l1_gas_price,
         l2_fair_gas_price: payload.l2_fair_gas_price,
         fair_pubdata_price: payload.fair_pubdata_price,
+        pubdata_params: payload.pubdata_params.unwrap_or_default(),
```

Comment:
Dumb question: Is the default value reasonable here (ditto for JSON-RPC-based syncing) because it's unused before the protocol version upgrade, at which point it's expected to be returned by the main node / peers? If the default value is read, it could silently produce unexpected results. So just from this perspective, it'd be safer to use an `Option` here to ensure that the value is indeed never read.